janus_core/helpers/post_process.py

```diff
@@ -0,0 +1,211 @@
+                Make partial RDFs. If `by_elements` is true will filter to
+                only display pairs in list.
+            _ : Any
+                Dummy to preserve interface with modern Analysis mode.
+            """
+            return get_rdf(data, rmax=rmax, nbins=nbins, elements=elements)
+
+    def dump_rdf(full_rdf: NDArray[float64], filename: PathLike) -> None:
+        """
+        Write RDF to file.
+
+        Parameters
+        ----------
+        full_rdf : NDArray[float64]
+            RDF to dump to file.
+        filename : PathLike
+            File to dump to.
+        """
+        for (rdfs, dists), out_path in zip(full_rdf, filename):
+            with open(out_path, "w", encoding="utf-8") as out_file:
+                for rdf, dist in zip(rdfs, dists):
+                    print(dist, rdf, file=out_file)
+
+    if not isinstance(data, Sequence):
+        data = (data,)
+
+    if by_elements:
+        elements = set(data[0].get_chemical_symbols()) if elements is None else elements
+
+        rdf = {
+            element: calc_rdf(rmax, nbins, element, index)
+            for element in combinations(elements, 2)
+        }
+    else:
+        rdf = calc_rdf(rmax, nbins, elements, index)
+
+    if filename is not None:
+        if not isinstance(filename, Sequence):
+            filename = (filename,)
+
+        if by_elements:
+            for full_rdf in rdf.values():
+                dump_rdf(full_rdf, filename)
+        else:
+            dump_rdf(rdf, filename)
+
+    return rdf
+
+
+def compute_vaf(
+    data: Sequence[Atoms],
+    filename: Optional[PathLike] = None,
+    *,
+    use_velocities: bool = False,
+    fft: bool = False,
+) -> NDArray[float64]:
+    """
+    Compute the velocity autocorrelation function (VAF) of `data`.
+
+    Parameters
+    ----------
+    data : Sequence[Atoms]
+        Dataset to compute VAF of.
+    filename : Optional[PathLike]
+        If present, dump resultant VAF to file.
+    use_velocities : bool
+        Compute VAF using velocities rather than momenta.
+    fft : bool
+        Compute the fourier transformed VAF.
+
+    Returns
+    -------
+    NDArray[float64]
+        Computed VAF.
+    """
+    if use_velocities:
+        momenta = np.asarray([datum.get_momenta() / datum.get_masses() for datum in data])
+    else:
+        momenta = np.asarray([datum.get_momenta() for datum in data])
+
+    n_steps = len(momenta)
+    n_atoms = len(momenta[0])
+
+    vaf = np.sum(
+        np.asarray(
+            [[np.correlate(momenta[:, j, i], momenta[:, j, i], "same")
+             for i in range(3)]
+             for j in range(n_atoms)]
+        ),
+        axis=1,
+    )
+
+    vaf /= n_steps - np.arange(n_steps)
+
+    if fft:
+        vaf = np.fft.fft(vaf)
+
+    if filename is not None:
+        with open(filename, "w", encoding="utf-8") as out_file:
+            for elem in vaf:
+                print(elem, file=out_file)
```

Comment:
We may want some kind of binary format option here, but maybe that can come later

Reply:
probably we shall see h5md approach and use it... Jacob is already aware.